sidecar/src/interface.rs

```diff
@@ -1000,6 +1094,128 @@ impl SidecarServer {
         let data = SendData::new(size, payload, headers, target);
         self.trace_flusher.enqueue(data);
     }
+
+    pub async fn compute_stats(&self) -> SidecarStats {
+        let mut telemetry_stats_errors = 0;
+        let telemetry_stats = join_all({
+            let sessions = self.sessions.lock().unwrap();
+            let mut futures = vec![];
+            for (_, s) in sessions.iter() {
+                let runtimes = s.runtimes.lock().unwrap();
+                for (_, r) in runtimes.iter() {
+                    let apps = r.apps.lock().unwrap();
+                    for (_, a) in apps.iter() {
+                        if let Some(Some(existing_app)) = a.peek() {
+                            match existing_app.telemetry.stats() {
+                                Ok(future) => futures.push(future),
+                                Err(_) => telemetry_stats_errors += 1,
+                            }
+                        }
+                    }
+                }
+            }
+            futures
+        })
+        .await;
+        let sessions = self.sessions.lock().unwrap();
+        SidecarStats {
+            trace_flusher: self.trace_flusher.stats(),
+            sessions: sessions.len() as u32,
+            session_counter_size: self.session_counter.lock().unwrap().len() as u32,
+            runtimes: sessions
+                .values()
+                .map(|s| s.runtimes.lock().unwrap().len() as u32)
+                .sum(),
+            apps: sessions
+                .values()
+                .map(|s| {
+                    s.runtimes
+                        .lock()
+                        .unwrap()
+                        .values()
```

Comment:
can we tell the formatter somehow that lock().unwrap().values() is perfectly readable on one line :-(

Reply:
As far as I know, it doesn't can't distinguish between different methods, so I guess no.

Reply:
I think that the repeated use of `lock().unwrap` should be hidden behind helper functions on this form to lessen the noise.
```
    fn runtimes(&self) -> MutexGuard<HashMap<String, RuntimeInfo>> {
        self.runtimes.lock().unwrap()
    }
```

Can be done for `sessions`, `runtimes`, `apps,` `app_or_actions` to make things at least a little bit nicer to read.

Reply:
Good idea, though I'd call it lock_runtimes() to be explicit about the locking.